lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -94,6 +89,11 @@ defmodule Mix.Tasks.Escript.Build do
     * `:emu_args` - emulator arguments to embed in the escript file.
       Defaults to `""`.
 
+    * `:include_priv_for` - a list of application names (atoms) specifying
+      applications which priv directory should be included in the resulting
+      escript archive. Currently the expected way of accessing priv files
+      in an escript is via `:escript.extract/2`. Defaults to `[]`.
```

Comment:
We could also be more strict and say that they should not rely on the exact archive layout, instead they should do something like this:

```
priv_path = :code.priv_dir(:app_name)
escript_path = Path.expand(:escript.script_name())
priv_in_archive = Path.relative_to(priv_path, escript_path)
```

(Ideally they would use `:erl_prim_loader`, but using it with escript is apparently deprecated, because it will change in the future)

Reply:
FTR the deprecation note for escript + `:erl_prim_loader` was added in https://github.com/erlang/otp/pull/8091.